server/src-lib/Hasura/Tracing.hs

```diff
@@ -76,9 +79,10 @@ instance HasReporter m => HasReporter (ExceptT e m) where
 -- the active span within that trace, and the span's parent,
 -- unless the current span is the root.
 data TraceContext = TraceContext
-  { tcCurrentTrace  :: Word64
-  , tcCurrentSpan   :: Word64
-  , tcCurrentParent :: Maybe Word64
+  { tcCurrentTrace   :: BS.ByteString
+  , tcCurrentSpan    :: BS.ByteString
+  , tcCurrentParent  :: Maybe BS.ByteString
+  , tcCurrentSampled :: Maybe Bool
```

Comment:
Are we going to require this field.

The sampled header is used to describe if the particular trace request needs to be traced or not. The value for this is either 0 or 1. This is an optional header. Do we drop this?

If we don't drop this, then should the interface be changed to:
```haskell
-- | Perform HTTP request which supports Trace headers
tracedHttpRequest 
  :: MonadTrace m 
  =>  HTTP.Request
  -- ^ http request that needs to be made
  -> Bool
  -- ^ Should the trace request be traced or not  
  -> (HTTP.Request -> m a)
  -- ^ a function that takes the traced request and executes it
  -> m a

Reply:
Since `Sampled` header is Optional and will be sent/used by external monitoring engines and also since we won't be processing this header, it's better we leave it out for now.